webdriver/tests/support/helpers.py

```diff
@@ -82,6 +82,40 @@ def _restore_windows(session):
     _switch_to_top_level_browsing_context(session)
 
 
+async def cleanup_bidi_session(session):
+    """
+    Clean-up the current BiDi session for a clean state.
+    """
+
+    async def _restore_browsing_contexts(session):
+        """
+        Close superfluous browsing contexts opened by the test.
+
+        If classic `session.window_handle` is set and is open, keep it open. Otherwise, keep open the first browsing
+        context, and set `session.window_handle` to it.
+        """
+        command_result_future = await session.bidi_session.send_command("browsingContext.getTree", {"maxDepth": 1})
+        command_result = await command_result_future
+        # List of top-level browsing contexts.
+        top_contexts = [c["context"] for c in command_result["contexts"]]
+
+        # If classic session has a window handle, and if it is still open, keep it open.
+        # Otherwise, keep open the first browsing context.
+        if session.window_handle is not None \
+                and session.window_handle in top_contexts:
+            current_window = session.window_handle
+        else:
+            current_window = top_contexts[0]
+            session.window_handle = current_window
+
+        for context in command_result["contexts"]:
```

Comment:
Above you created a dedicated variable. Please fix all instances of it:

```suggestion
        for context in top_contexts:
```